### DISCUSSION TITLE
Deluge Slow speed

### DISCUSSION THREAD

**Discussion by 00xCC** [qdm12/gluetun#1816] [Q&A]
Hello, 

I'm currently using Surfshark VPN and have lately begun utilizing Gluten for my ARR setup. I tried to deploy qBittorrent before trying deluge, but it never worked, so I switched to the deluge docker client. Overall, I like this project, however whenever I try to download a torrent, it either gets stuck at 99% or never gets enough peers to complete the download. Is anyone else having the same problem with this setup and know how to fix it?

**Comment by gmillerd** [comment]
This seems to be a torrent client issue, not a gluetun issue? The container nor the services there in know a torrent protocol from any other protocol.

**Reply by 00xCC** [reply]
Thanks. I assumed the problem was with Deluge, but when I tried to download directly on my PC using the VPN configuration and Deluge client, I got maximum speed, so I'm not sure what the real problem is. Is there a method to fix the qBittorrent docker issue? I tried to deploy the linuxserver image, but the download is stalling at the metadata check.

**Reply by gmillerd** [reply]
Speed issues and torrent's completing ... very separate issues?

It sounds like a lot of stuff is doing on here, I would be reviewing that the egress of bt is actually gluetun, this can be verified by you and again by posting the docker-compose (as you see a lot of people do in these issues). 

When I see "download stalling on metadata check" its not clear what you mean, torrent on "downloading metadata"?

qbt and deluge do a good job of telling you they are blocked by firewalls (or ingress/egress issues) ... if you aren't even using gluetun or not port forwarding (which is what i suspect) ... you are not going to have good performance and other peers are going to not like transferring to you.

In the end, easily verifiable. This is just a containerized vpn/gateway services, not a bittorrent client.

**Reply by 00xCC** [reply]
Sorry for the confusion, actually before trying deluge I went ahead with qBittorrent but got into torrent stall issue which is mentioned in this link - https://github.com/qdm12/gluetun/discussions/1562.

Here is the compose file for my current docker deployment,

version: "3"
services:
  gluetun:
    image: qmcgaw/gluetun
    cap_add:
      - NET_ADMIN
    environment:
      - VPN_SERVICE_PROVIDER=surfshark
      - VPN_TYPE=wireguard
      - WIREGUARD_PRIVATE_KEY= <key>
      - WIREGUARD_ADDRESSES=10.10.0.2/16
      - SERVER_COUNTRIES= <country name>
    ports:
      - "0.0.0.0:8112:8112/tcp"  
      - 6881:6881
      - 6881:6881/udp
restart: unless-stopped
  deluge:
    image: lscr.io/linuxserver/deluge:latest
    container_name: deluge
    network_mode: "service:gluetun" 
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Etc/UTC
      - DELUGE_LOGLEVEL=error #optional
    volumes:
- /path/qbittorrent:/config 
- /path/Downloads:/downloads
    depends_on:
      - gluetun
    restart: unless-stopped


Let me know if this compose file is correct or should i make any changes to it.

**Reply by 00xCC** [reply]
@gmillerd just wanted to confirm if this docker compose file looks okay or should I make any change to it. Also, regarding firewall rules, I'm not hosting it behind any firewall, so do I still need to make any changes on my router in terms of opening the ports?
